pkg/kp/healthmanager.go

```diff
@@ -332,9 +385,29 @@ func processHealthUpdater(
 }
 
 // Helper to processHealthUpdater()
-func healthEquiv(x *WatchResult, y *WatchResult) bool {
-	return x == nil && y == nil ||
-		x != nil && y != nil && x.ValueEquiv(*y)
+func healthEquiv(x *WatchResult, y *WatchResult, tokens int64) bool {
+	if x == nil && y == nil {
+		return true
+	}
+	if !(x != nil && y != nil) {
+		return false
+	}
+	if tokens < OutputLimit {
+		// If the health check is updating Consul frequently, only update when absolutely
+		// necessary.
+		return x.Status == y.Status
+	}
+	return x.ValueEquiv(*y)
+}
+
+func toThrottled(wr *WatchResult) *WatchResult {
+	return &WatchResult{
+		Node:    wr.Node,
+		Id:      wr.Id,
+		Service: wr.Service,
+		Status:  "throttled",
```

Comment:
Please make sure this is enumerated with the other health states. That said, I really like the idea of a state just intended to represent service flapping.